api/app/serializers/v1/project_collection_serializer.rb

```diff
@@ -5,28 +5,35 @@ class ProjectCollectionSerializer < ManifoldSerializer
 
     abilities
 
-    typed_attribute :title, NilClass
-    typed_attribute :slug, NilClass
-    typed_attribute :pending_slug, NilClass
-    typed_attribute :sort_order, NilClass
-    typed_attribute :visible, NilClass
-    typed_attribute :homepage, NilClass
-    typed_attribute :position, NilClass
-    typed_attribute :icon, NilClass
-    typed_attribute :number_of_projects, NilClass
-    typed_attribute :featured_only, NilClass
-    typed_attribute :smart, NilClass
-    typed_attribute :description, NilClass
-    typed_attribute :description_formatted, NilClass
-    typed_attribute :description_plaintext, NilClass
-    typed_attribute :sort_order, NilClass
-    typed_attribute :tag_list, NilClass
-    typed_attribute :description, NilClass
-    typed_attribute :homepage_start_date, NilClass
-    typed_attribute :homepage_end_date, NilClass
-    typed_attribute :homepage_count, NilClass
-    typed_attribute :manually_sorted, NilClass, &:manually_sorted?
-    typed_attribute :projects_count, NilClass, &:collection_projects_count
+    typed_attribute :title, Types::String.meta(unique: true)
+    typed_attribute :slug, Types::String.meta(description: I18n.t("attributes.descriptions.slug"), read_only: true)
+    typed_attribute :pending_slug, Types::String.meta(description: I18n.t("attributes.descriptions.pending_slug"))
+    typed_attribute :sort_order, Types::String.enum(
+      "created_at_asc",
+      "created_at_desc",
+      "updated_at_asc",
+      "updated_at_desc",
+      "publication_date_asc",
+      "publication_date_desc",
+      "title_asc",
+      "title_desc"
+    )
+    typed_attribute :visible, Types::Bool
+    typed_attribute :homepage, Types::Bool
+    typed_attribute :position, Types::Integer
+    typed_attribute :icon, Types::String.enum("lamp")
+    typed_attribute :number_of_projects, Types::Integer.optional
+    typed_attribute :featured_only, Types::Bool
+    typed_attribute :smart, Types::Bool
+    typed_attribute :description, Types::String.optional
+    typed_attribute :description_formatted, Types::String.meta(read_only: true)
+    typed_attribute :description_plaintext, Types::String.meta(read_only: true)
+    typed_attribute :tag_list, Types::Array.of(Types::String) # check value
+    typed_attribute :homepage_start_date, Types::String.optional
+    typed_attribute :homepage_end_date, Types::String.optional
+    typed_attribute :homepage_count, Types::Integer.optional
+    typed_attribute :manually_sorted, Types::Bool.meta(read_only: true), &:manually_sorted?
+    typed_attribute :projects_count, Types::Integer.meta(read_only: true), &:collection_projects_count
 
     typed_has_many :collection_projects, paginated: true do |object, params|
```

Comment:
It appears that this `projects_collection` attribute  should be paginated in it's response. I don't see a current response that looks like the project_collections are paginated. 

When I made a PATCH request to the `project_collections/:id` route 4 months ago, I got this back as a part of the response:

```
"relationships": {
      "collectionProjects": {
        "data": [
          {
            "id": "6aaaca05-08f6-4e7c-935f-ab6066226d2b",
            "type": "collectionProjects"
          }
        ],
        "links": {
          "first": "/api/v1/project_collections/aa9e7761-8c28-4b6e-a927-7575c6913f14/relationships/collection_projects?page=1&per_page=25",
          "last": "/api/v1/project_collections/aa9e7761-8c28-4b6e-a927-7575c6913f14/relationships/collection_projects?page=1&per_page=25"
        },
        "meta": {
          "pagination": {
            "currentPage": 1,
            "nextPage": 0,
            "prevPage": 0,
            "totalPages": 1,
            "totalCount": 1,
            "perPage": 25
          }
        }
      },
      "subjects": {
        "data": []
      },
      "projects": {
        "data": [
          {
            "id": "841da660-4eb1-4091-af11-0ab14c7c4b61",
            "type": "projects"
          }
        ]
      }
    }
```
and when I make the request to the same route today, the relationships response looks like:

```
 "relationships": {
      "collectionProjects": {
        "data": [
          {
            "id": "39ef5d99-075e-4100-9323-cd6244b727f6",
            "type": "collectionProjects"
          }
        ]
      },
      "subjects": {
        "data": []
      }
    },
    "meta": {
      "partial": false
    }
```

I feel like this could take me down a rabbit hole, so I wanted to check in with you before I started digging deeper.

Reply:
Yeah, don't dig deeper. I will take a look at this. Some things changed around this route as part of 8db1480a8cca30b735b4dde4dce523e4eea9f0a4, and I can circle back and see what happened here.